src/app/services/store.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { combineLatest, map, Observable } from 'rxjs';
+import { StoreModel } from '../models/store.model';
+import { StoreTagModel } from '../models/store-tag.model';
+
+@Injectable({ providedIn: 'root' })
+export class StoreService {
+  constructor(private _httpClient: HttpClient) {
+  }
+
+  getAll(): Observable<StoreModel[]> {
+    return this._httpClient.get<StoreModel[]>('https://6384fca14ce192ac60696c4b.mockapi.io/freshcart-stores');
+  }
+
+  getAllWithTags(): Observable<StoreModel[]> {
+    return combineLatest([
+      this._httpClient.get<StoreModel[]>('https://6384fca14ce192ac60696c4b.mockapi.io/freshcart-stores'),
```

Comment:
because you make two calls to HTTP client, i would keep my service more stupid and move this logic to component